### DISCUSSION TITLE
[Idea] There should be free delivery calculation options - ONLY_PRODUCTS or BOTH_WITHOUT_SHIPPING

### DISCUSSION THREAD

**Discussion by lim3ra** [PrestaShop/PrestaShop#33777] [Ideas]
### The problem

Currently, Prestashop calculates free delivery based on total order value, including coupons.

classes\Cart.php
`$orderTotalwithDiscounts = $this->getOrderTotal(true, Cart::BOTH_WITHOUT_SHIPPING, null, null, false);`

Example scenario:
- Delivery cost is $20
- Free delivery is set to $100
- Customer adds to cart product for $100
- Free delivery is applied
- Customer adds $10 coupon to cart
- Total order is now $90 so free delivery does not apply anymore and $20 is added so the final cost is $110 which is not fair from customer's perspective

### The solution

The solution is to change the way free delivery is calculated, for this scenario it should be product price, excluding coupons. 

classes\Cart.php
`$orderTotalwithDiscounts = $this->getOrderTotal(true, Cart::ONLY_PRODUCTS, null, null, false);`

And we should have the option to change this behavior in the back office, under free delivery preferences. 

It should look like this:
![Screenshot 2023-08-30 205029](https://github.com/PrestaShop/PrestaShop/assets/26142543/81c81d08-8238-42a9-8f0e-1c2cc1f76597)

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by RosaBenouamer** [comment]
Interesting as point of view :-)
